### PR TITLE
Remove third party types from public APIs

### PR DESCRIPTION
**Pull request by ysaito1001**
## Motivation and Context
Addresses item 1, 3, and 9 in #2413 

## Description
This PR removes the third party types as follows:
- removes `InvalidHeaderValue` from public API in `aws-http`
- removes `SendError` from public API in `aws-smithy-async`
- removes `xmlparser` from public API in `aws-smithy-xml`

Those types were exposed to public APIs primarily due to the implementation of traits, e.g. `From` and `Iterator`. Those implementations are used internally (such as XML deserialization and converting an error type) so we should be able to hide them within crates.

## Testing
- [x] Passed tests in CI

## Checklist

- [x] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates
- [x] I have updated `CHANGELOG.next.toml` if I made changes to the AWS SDK, generated SDK code, or SDK runtime crates

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Review by Velfi — APPROVED** [review]

File: rust-runtime/aws-smithy-async/src/future/rendezvous.rs [hunk 1]
```diff
@@ -38,6 +37,35 @@ pub fn channel<T>() -> (Sender<T>, Receiver<T>) {
...
 }
 
+/// Errors for rendezvous channel
+pub mod error {
+    use std::fmt;
+
+    /// Error when [crate::future::rendezvous::Sender] fails to send a value to the associated `Receiver`
+    #[derive(Debug)]
+    pub struct SendError<T> {
+        source: tokio::sync::mpsc::error::SendError<T>,
```
Comment (jdisanti) [inline rust-runtime/aws-smithy-async/src/future/rendezvous.rs:47]: Nit: import things when not in codegen =)

File: rust-runtime/aws-smithy-xml/src/decode.rs [hunk 2]
```diff
@@ -256,6 +254,13 @@ impl<'inp> Document<'inp> {
     }
 }
 
+/// A new-type wrapper around `Token` to prevent the wrapped third party type from showing up in
+/// public API
+#[derive(Debug)]
+pub struct XmlToken<'inp> {
+    token: Token<'inp>,
+}
```
Comment (jdisanti) [inline rust-runtime/aws-smithy-xml/src/decode.rs:262]: Nit: might be slightly easier to read as `XmlToken<'inp>(Token<'inp>)`

**Review by jdisanti — APPROVED** [review]
This is great! Thanks.

File: rust-runtime/aws-smithy-async/src/future/rendezvous.rs (hunk 1 above)
Comment (ysaito1001) [reply rust-runtime/aws-smithy-async/src/future/rendezvous.rs]: Used an alias to bind the import to `TokioSendError` in 59ffe31.

File: rust-runtime/aws-smithy-xml/src/decode.rs (hunk 2 above)
Comment (ysaito1001) [reply rust-runtime/aws-smithy-xml/src/decode.rs]: Agreed. Addressed in 3105f8f.

**Pull request merged**